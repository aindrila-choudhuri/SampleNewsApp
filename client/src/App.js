import './App.css';
import NewsList from "./components/NewsList"

function App() {
  return (
    <div data-testid="app" className="App">
        <NewsList />
    </div>
  );
}

export default App;
