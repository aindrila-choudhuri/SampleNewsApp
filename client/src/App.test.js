import React from 'react';
import App from "./App";
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

it("renders App component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
});

it("renders App component correctly", () => {
    const {getByTestId} = render(<App />);
    const app = getByTestId('app');
    
    expect(app).toContainElement(app);
    expect(app).toHaveClass("App");
});

it("app component matches snapshot", () => {
    const domTree = renderer.create(<App />).toJSON();
    expect(domTree).toMatchSnapshot();
});