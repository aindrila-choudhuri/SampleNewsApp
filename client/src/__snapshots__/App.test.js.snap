// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component matches snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <div
    data-testid="newslist"
  >
    <div
      className="topnav"
    >
      <h1
        className="spanheading"
      >
        Latest News
      </h1>
      <div
        className="search-container"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
        >
          <i
            className="fa fa-search"
          />
        </button>
      </div>
    </div>
    <div
      className="all__news"
      data-testid="allNews"
    />
  </div>
</div>
`;
